lib/mapping/index.d.ts

```diff
@@ -128,27 +128,27 @@ export namespace mapping {
     update(doc: any, docInfo?: UpdateDocInfo): ModelBatchItem;
   }
 
-  interface ModelMapper {
+  interface ModelMapper<T = { [key: string]: any }> {
```

Comment:
I think we should continue to use `any` that is less restrictive than `{ [key: string]: any }`.

Reply:
I can't think of any case where you need any.
If you want backwards compatibility, we can just change it to:
`interface ModelMapper<T = any }> {`

Reply:
I prefer not to be restrictive, users can use it for whatever type they choose, but tbh I don't know which cases are not going to be covered by `T = { [key: string]: any }`

Reply:
We should use `T = { [key: string]: any }` because the driver always returns null or an object for .get() and equivalent for .find(). 
We are not being restrictive by not using `any`. We are just doing what typescript is intended for in my opinion :)

Reply:
I think its OK to use `{ [key: string]: any }`, my concern is that it was already implemented using `any`, so restricting it now might be a breaking change.

Let's continue addressing the other issues (like `ModelMapperGeneric<T>`) and come back to this one if needed.

Reply:
ModelMapperGeneric and this point is the same in my opinion.
If we use `any` it will be fully backwards compatible and can keep ModelMapper as the only interface. Best solution right now I think.

Reply:
Backwards compatibility is very important as a library.

Note that if somebody is using the following:

```typescript
const modelMapper: ModelMapper = mapper.forModel(x);
```

With this new change, it will break.

Would it be possible to provide additional generic interfaces and make the existing ones work as currently do?

Reply:
this is the least I can do to contribute to this great library!